scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/simulation/SimulationFacade.java

```diff
@@ -232,7 +232,11 @@ private ActivityDirective schedulingActToActivityDir(SchedulingActivityDirective
       final var durationType = activity.getType().getDurationType();
       if (durationType instanceof DurationType.Controllable dt) {
         arguments.put(dt.parameterName(), SerializedValue.of(activity.duration().in(Duration.MICROSECONDS)));
-      } else if (durationType instanceof DurationType.Uncontrollable || durationType instanceof DurationType.Fixed) {
+      } else if (
+          durationType instanceof DurationType.Uncontrollable
+          || durationType instanceof DurationType.Fixed
+          || durationType instanceof DurationType.Parametric
+      ) {
         // If an activity has already been simulated, it will have a duration, even if its DurationType is Uncontrollable.
       } else {
         throw new Error("Unhandled variant of DurationType: " + durationType);
```

Comment:
Is this code reachable if the DurationType interface is sealed?

Reply:
This is a precaution in case new types are added, requiring this to be updated. We should exhaustively check all possibilities, but Java doesn't require exhaustive switch statements (YET). So the alternative is to throw an error by default.